### PR TITLE
feat(ng): add warning when angular-cli is detected locally

### PR DESCRIPTION
**Pull request by beeman**

### CODE REVIEW

File: packages/@angular/cli/bin/ng [hunk 1]
```diff
@@ -86,6 +86,34 @@ if (require('../package.json')['name'] == 'angular-cli'
   `));
 }
 
+const packageJsonProject = require(path.join(process.cwd(), 'package.json'));
+const hasOldDep = Boolean(packageJsonProject.dependencies['angular-cli']);
```
Comment (hansl) [inline packages/@angular/cli/bin/ng]: `!!` is preferred.

**Comment by hansl** [comment]
cc @StephenFluin @robwormald

LGTM outside of the nit above.

**Comment by beeman** [comment]
Nit addressed

**Comment by beeman** [comment]
WIP: Please don't merge yet.

I just found out that this causes an error when running it outside of a project:

```
$ cd
$ ng version
module.js:472
    throw err;
    ^

Error: Cannot find module '/Users/beeman/package.json'
    at Function.Module._resolveFilename (module.js:470:15)
    at Function.Module._load (module.js:418:25)
    at Function.Module._load (/Users/beeman/workspace/angular-cli/angular-cli/lib/bootstrap-local.js:69:24)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/beeman/workspace/angular-cli/angular-cli/packages/@angular/cli/bin/ng:89:28)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)

```

I will add a check for this.

**Comment by beeman** [comment]
Issues should be fixed, @hansl can you please re-review?

File: packages/@angular/cli/bin/ng [hunk 2]
```diff
@@ -86,6 +86,30 @@ if (require('../package.json')['name'] == 'angular-cli'
   `));
 }
 
+const packageJsonProjectPath = path.join(process.cwd(), 'package.json')
```
Comment (hansl) [inline packages/@angular/cli/bin/ng]: We allow people to run anywhere inside a project. You should use the `__dirname` instead (and potentially `findup`).

**Pull request closed without merging**